app/controllers/api/v1/user_annotations_controller.rb

```diff
@@ -101,7 +101,7 @@ def create
           ErrorTracker.report_exception(e, current_user, @study, log_params)
           render json: {error: e.message}, status: 400 # Bad request
         rescue => e
-          ErrorTracker.report_exception(e, current_user, error_context)
+          ErrorTracker.report_exception(e, current_user, @study, log_params)
```

Comment:
This change follows the same convention as the other error report so it seems like an appropriate move 👍 however I have a Rails question:
I thought that instance variables defined with the `@` sign were to be defined in a Controller to then be useable in a View, however I don't see a definition/assignment of `@study` in this Controller class. Could you explain where `@study` gets assigned? Thanks!

Reply:
Great question!  This is definitely a Rails-specific thing.  Controllers can have helper methods that fire before the actual requested method is executed via the `before_action` declaration.  These are usually setters, getters, or authentication methods, although you can do anything you want, and any instance variables you define will then be available in the actual method.  In this case, the we're calling the [`set_study`](https://github.com/broadinstitute/single_cell_portal_core/blob/development/app/controllers/api/v1/concerns/study_aware.rb#L7) method with this line:

```
include Concerns::StudyAware
... 
before_action :set_study
```
In Ruby, calling `include` will make any instance methods from a module available in the class you're calling `include` from, and `set_study` is defined in `app/controllers/api/v1/concerns/study_aware.rb` as this:

```
def set_study
  @study = Study.any_of({accession: params[:study_id]},{id: params[:study_id]}).first
  if @study.nil? || @study.queued_for_deletion?
    head 404 and return
  end
end
```

Reply:
ah okay great, thanks!